### PR TITLE
Bbrooks/bluebutton1313 RDS User Password Change Playbook

### PR DESCRIPTION
**Pull request by cthulhuplus**
**Why**
Change the RDS passwords so that they are no long the shared HealthApt passwords

**What Changed**
Created an Ansible playbook that will update the password for one of the pre-existing RDS users.

**Testing**
Ran the playbook successfully against a test environment.

**Security Impact**
Changes the passwords being used for App communication to the RDS instances.

### CODE REVIEW

File: ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#ansible-playbook rotate_bfd_postgres_user.yml --extra-vars "rds_database= master_user= master_pw= old_user= db_user= db_user_pw= rds_region="
+- hosts: localhost
+  vars:
+    rds_database: 
+    master_user:
```
Comment (jzulim) [inline ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml]: Maybe consider storing these details (master user and pass) in the respective environment vault files and pull from there. Just a thought.

File: ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#ansible-playbook rotate_bfd_postgres_user.yml --extra-vars "rds_database= master_user= master_pw= old_user= db_user= db_user_pw= rds_region="
```
Comment (jzulim) [inline ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml]: How is the old_user var utilized here?

File: ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+      register: database_info
+
+    - name: Change Pre-existing Users Password
+      postgresql_user:
+        login_host: "{{ database_info.instances[0].endpoint.address }}"
+        login_user: "{{ database_info.instances[0].master_username }}"
+        login_password: "{{ master_pw }}"
+        db: postgres
+        name: "{{ db_user }}"
```
Comment (jzulim) [inline ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml]: I don't see a password being set. Maybe I'm missing something, but I think we need a `password` field specified.

File: ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml (hunk 2 above)
Comment (cthulhuplus) [reply ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml]: The old_user variable was supposed to dropped, I switched to db_user as I felt like it made more logical sense.

File: ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml (hunk 1 above)
Comment (cthulhuplus) [reply ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml]: I was under the impression that you could either pass them in via the command line or a vars file and Ansible would handle them either way. If that is not the case I definitely want to make sure it supports Ansible Vault. I'll follow up with you.

Comment (jzulim) [reply ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml]: You're correct, either way does work. I was thinking, I dont see this master user and pass stored in the vaults currently (would be good to add them for safe keeping) and then you could pull these values from there.

File: ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml (hunk 3 above)
Comment (cthulhuplus) [reply ops/ansible/playbooks-ccs/rotate_bfd_postgres_user.yml]: That's weird, I must've accidentally cut too much during refactoring. I added it back.

**Comment by cthulhuplus** [comment]
Test in a separate AWS Account:
ansible-playbook rotate_bfd_postgres_user.yml -e "ansible_python_interpreter=/usr/bin/python3" --extra-vars "@~/rds_file.txt"
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAY [localhost] *********************************************************************************************************************************************************************************************************************************************************************************************************
TASK [Gathering Facts] ***************************************************************************************************************************************************************************************************************************************************************************************************ok: [localhost]

TASK [RDS Gather Facts/Info] *********************************************************************************************************************************************************************************************************************************************************************************************[DEPRECATION WARNING]: The 'rds_instance_facts' module has been renamed to 'rds_instance_info'. This feature will be removed in version 2.13. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
ok: [localhost]

TASK [Show Register] *****************************************************************************************************************************************************************************************************************************************************************************************************ok: [localhost] => {
    "database_info": {
        "changed": false, 
        "deprecations": [
            {
                "msg": "The 'rds_instance_facts' module has been renamed to 'rds_instance_info'", 
                "version": "2.13"
            }
        ], 
        "failed": false, 
        "instances": [
            {
                "allocated_storage": 10, 
                "associated_roles": [], 
                "auto_minor_version_upgrade": false, 
                "availability_zone": "us-east-2a", 
                "backup_retention_period": 1, 
                "ca_certificate_identifier": "rds-ca-2019", 
                "copy_tags_to_snapshot": false, 
                "db_instance_arn": "arn:aws:rds:us-east-2:666435484074:db:bbrooks-database", 
                "db_instance_class": "db.t3.small", 
                "db_instance_identifier": "bbrooks-database", 
                "db_instance_port": 0, 
                "db_instance_status": "available", 
                "db_parameter_groups": [
                    {
                        "db_parameter_group_name": "default.postgres11", 
                        "parameter_apply_status": "in-sync"
                    }
                ], 
                "db_security_groups": [], 
                "db_subnet_group": {
                    "db_subnet_group_description": "default", 
                    "db_subnet_group_name": "default", 
                    "subnet_group_status": "Complete", 
                    "subnets": [
                        {
                            "subnet_availability_zone": {
                                "name": "us-east-2b"
                            }, 
                            "subnet_identifier": "subnet-b77bbfcd", 
                            "subnet_status": "Active"
                        }, 
                        {
                            "subnet_availability_zone": {
                                "name": "us-east-2c"
                            }, 
                            "subnet_identifier": "subnet-010c3f4c", 
                            "subnet_status": "Active"
                        }, 
                        {
                            "subnet_availability_zone": {
                                "name": "us-east-2a"
                            }, 
                            "subnet_identifier": "subnet-ef29a587", 
                            "subnet_status": "Active"
                        }
                    ], 
                    "vpc_id": "vpc-276de64f"
                }, 
                "dbi_resource_id": "db-M3ZKXVKBBFILW3AIO4IJU2VWMA", 
                "deletion_protection": false, 
                "domain_memberships": [], 
                "endpoint": {
                    "address": "bbrooks-database.czprajbwowl8.us-east-2.rds.amazonaws.com", 
                    "hosted_zone_id": "Z2XHWR1WZ565X2", 
                    "port": 5432
                }, 
                "engine": "postgres", 
                "engine_version": "11.5", 
                "iam_database_authentication_enabled": false, 
                "instance_create_time": "2019-11-08T14:51:30.858000+00:00", 
                "latest_restorable_time": "2020-02-20T15:12:08+00:00", 
                "license_model": "postgresql-license", 
                "master_username": "bbrooks_admin", 
                "monitoring_interval": 0, 
                "multi_az": false, 
                "option_group_memberships": [
                    {
                        "option_group_name": "default:postgres-11", 
                        "status": "in-sync"
                    }
                ], 
                "pending_modified_values": {}, 
                "performance_insights_enabled": false, 
                "preferred_backup_window": "08:16-08:46", 
                "preferred_maintenance_window": "mon:07:14-mon:07:44", 
                "publicly_accessible": true, 
                "read_replica_db_instance_identifiers": [], 
                "storage_encrypted": false, 
                "storage_type": "gp2", 
                "tags": {
                    "Application": "testing", 
                    "Environment": "testing"
                }, 
                "vpc_security_groups": [
                    {
                        "status": "active", 
                        "vpc_security_group_id": "sg-0bbcbee327a41a12b"
                    }, 
                    {
                        "status": "active", 
                        "vpc_security_group_id": "sg-9a5820f1"
                    }
                ]
            }
        ]
    }
}

TASK [Change postgres RDS instance password] *****************************************************************************************************************************************************************************************************************************************************************************changed: [localhost]

PLAY RECAP ***************************************************************************************************************************************************************************************************************************************************************************************************************localhost                  : ok=4    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

**Comment by cthulhuplus** [comment]
After discussion with John Zulim and Nathan Dister, this was scoped down to just cover the bb user as there are additional improvements needed for the etl user. I have tested the change in TEST, logged into the test RDS database with the new bb user and password, queried tables it is supposed to have access to, and then disabled the original user.

Command I use to run the playbook:

ansible-playbook rotate_bfd_postgres_user.yml -vvv -e "ansible_python_interpreter=/usr/bin/python3" --extra-vars "@vars/test/group_vars/all/vault.yml" --extra-vars '{ "rds_region":"us-east-1", "rds_database":"bfd-test-master" }' --ask-vault-pass

Then I manually logged into the database and ran:

ALTER USER svc_fhir_bb WITH NOLOGIN;

**Comment by cthulhuplus** [comment]
Created ticket https://jira.cms.gov/browse/BLUEBUTTON-2000 to capture the etl user work

**Pull request closed without merging**